driver-core/src/test/java/com/datastax/driver/core/NetworkTopologyStrategyTest.java

```diff
@@ -571,6 +585,23 @@ public void networkTopologyStrategyExampleTopologyTest() {
     }
 
     @Test(groups = "unit")
+    public void networkTopologyStrategyNoNodesInDCTest() {
+        long t1 = System.currentTimeMillis();
+        Map<Token, Set<Host>> replicaMap =  networkTopologyStrategy(rf(DC1, 2), rf(DC2, 2))
+                .computeTokenToReplicaMap(largeRingTokenToPrimary, largeRing);
+        assertThat((System.currentTimeMillis() - t1) < 45000);
```

Comment:
Should be:

    assertThat(System.currentTimeMillis() - t1).isLessThan(45000);

Also, maybe we could lower the limit a bit. On my laptop, the optimized version consistently runs under a second.